### PR TITLE
add workshop - robot emails and templating

### PR DESCRIPTION
**Pull request by Rohan-Bansal**
Hi! I'm Rohan, and I just created a new workshop on sending robot emails with easy substitution!

#### *1-2 minute video clip from your test club meeting showing what members built*

See it here: https://cloud-kjphlxr7m.vercel.app/robot_vid.mp4 (also on slack #workshop-skunkworks)

#### *Quick Description*

This workshop is a quick and relatively simple (20-25 minutes max) project utilizing Python's SMTP library. It allows the user to create, send, and modify emails at a much, much faster rate to many more people; by using Python's native string formatting, it's quite modular too!

#### *Easy to hack on for long periods of time after completion*

There are a __lot__ of things you can do with this after completion; extensions can keep you occupied for weeks. A few examples of further projects include: switchable SMTP servers, switchable message bodies during runtime, auto responders, and many more.

#### *Have instructive and compelling images, GIFs, and visuals*

As a terminal application, there isn't much in the way of visuals, but I included everything I could to make sure the workshop is clear and compelling.

#### *Use repl.it or a similar service that allows members to begin coding with one click*

All the main code and demos are on [repl.it](https://repl.it) as well as on GitHub.

#### *Link unknown concepts to external sources*

Every single unknown concept in the tutorial is linked on the spot.

#### *End with at least 3 demos of how the project can be expanded, with live links and code links for each demo*

I added three demos, some of which were enumerated above. All have descriptions, live code demos, and a home on GitHub.

### CODE REVIEW

**Comment by Rohan-Bansal** [comment]
Please let me know if there's anything to change! I'd love the feedback.

**Comment by zachlatta** [comment]
Awesome! I'm assigning @MatthewStanciu to be your reviewer on this :-).

**Comment by Rohan-Bansal** [comment]
Hi! I understand there's a backlog, but I was wondering when I'd get an update. @MatthewStanciu ?

**Comment by Hugoyhu** [comment]
Hi, please provide a slack handle and/or hack club you lead, thanks!

**Comment by Rohan-Bansal** [comment]
@Hugoyhu I'm "solace" (Rohan Bansal) on Slack.

**Comment by Hugoyhu** [comment]
Thank you!

On Oct 7, 2020, at 10:58 PM, Rohan Bansal <notifications@github.com<mailto:notifications@github.com>> wrote:



@Hugoyhu<https://github.com/Hugoyhu> I'm "solace" (Rohan Bansal) on Slack.

**Review by MatthewStanciu — CHANGES_REQUESTED** [review]
Thanks so much for submitting this! This looks like a super cool workshop.

Here’s a list of thoughts I had as I was going through the workshop:

* Instead of “head over to repl.it and create a new Python3 project”, I would just say “Head over to repl.it/languages/python3
* Include an image or gif demonstrating how to upload files
* “The contacts.txt file” line includes a typo “bu”
* The \<Your Name\> with the <> formatting may be confusing to some people. I would remove the angle brackets there
* The _Quick Note_ thing should all be one line in italics
* It may be worth including a .env setup step so that people aren’t putting the password to their google account in a public repl
* Does this workshop work for school google accounts? If you have a school google account that you can use to verify this, you definitely should, because if it doesn’t, many club members will likely be confused
* Love that a lot of the information that would take a while to explain is opt-in and links to pages with more info if people want to know 👍
* Maybe before the “Getting Contacts” section, I would include a screenshot of you repl so that people have some context of where each piece of code is and where we currently are in the workshop. (It seems like a small thing but it’s really easy for people, especially beginners, to lose track of what’s going on in a workshop)
* A little unclear to me how much Python people are supposed to understand. Is this a beginner workshop? Intermediate? If there are any concepts people are supposed to understand before they get started, you should specify at the beginning.
* A lot of the code you write is presented in large chunks with little explanation (though the “important elements” part is good). You said not to copy but the more code is presented in large chunks the more likely people are to copy
* I got a little lost at first glance at the bottom of the “reading the message” section. 2 of the 3 code blocks starting after “Let’s go over:” are calling back to the previous chunk of code and explaining bits of it, while the third one is code that you’re supposed to add at the bottom of the file. Instead of “Test out both of these functions, here’s how to do it:” I would instead say “Let’s test both of these functions! At the end of your `main.py` file, add:”
* “After doing this, go ahead an remove the print statements” line is confusing to me. After doing what? Was I supposed to have run the file after I added those lines of code? This should be clearer
* Love the examples on how to expand the workshop at the end

**Comment by Rohan-Bansal** [comment]
Hi @MatthewStanciu ! I just tweaked the workshop to include all of your suggestions. About the school-email thing, it's definitely possible! I built/tested this workshop with my school account, and was able to follow all the steps.

Could you look over my changes?

File: workshops/robot_emails/README.md [hunk 1]
```diff
@@ -34,28 +35,31 @@ Head over to [this link](https://myaccount.google.com/u/0/lesssecureapps?hl=en)
...
 You will also need to complete [this captcha](https://accounts.google.com/b/0/DisplayUnlockCaptcha) to enable access for the next application that uses your credentials (this one).
 
-*Quick Notes*
+*Quick Notes*:
 
-If you happen to have 2 Factor Authentication enabled for your account, please follow [this article](https://support.google.com/accounts/answer/185833?hl=en) to generate an app password. When we write the program and input our account password, replace that with the app password you generated.
+*If you happen to have 2 Factor Authentication enabled for your account, please follow [this article](https://support.google.com/accounts/answer/185833?hl=en) to generate an app password. When we write the program and input our account password, replace that with the app password you generated.*
 
-After working on this project, it is recommended to turn the less secure apps lever back to it's off position.
+*After working on this project, it is recommended to turn the less secure apps lever back to it's off position.*
```
Comment (lachlanjc) [inline workshops/robot_emails/README.md:42]:
```suggestion
*After working on this project, it is recommended to turn the less secure apps lever back to its off position.*
```

**Comment by MatthewStanciu** [comment]
Wow, this is really good now! I think it’s basically ready to merge. I’ll do one more deep dive and let you know.

**Review by MatthewStanciu — APPROVED** [review]

**Review by lachlanjc — APPROVED** [review]

**Pull request merged**